src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/dao/HttpGoogleBillingDAO.scala

```diff
@@ -0,0 +1,337 @@
+import org.broadinstitute.dsde.workbench.model.{ErrorReport, ErrorReportSource, WorkbenchExceptionWithErrorReport}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+class HttpGoogleBillingDAO(appName: String, clientSecrets: GoogleClientSecrets, serviceAccountPemFile: String)
+                           (implicit val system: ActorSystem, val executionContext: ExecutionContext)
+  extends GoogleUtilities {
+
+  protected val workbenchMetricBaseName = "billing"
+  implicit val service = GoogleInstrumentedService.Iam
+
+  implicit val counters = googleCounters(service)
+  implicit val histo = googleRetryHistogram(service)
+
+  implicit val errorReportSource = ErrorReportSource("gpalloc-google")
+
+  val serviceAccountClientId: String = clientSecrets.getDetails.get("client_email").toString
+
+  lazy val httpTransport = GoogleNetHttpTransport.newTrustedTransport
+  lazy val jsonFactory = JacksonFactory.getDefaultInstance
+
+  //giant bundle of scopes we need
+  val saScopes = Seq(
+    StorageScopes.DEVSTORAGE_FULL_CONTROL,
+    ComputeScopes.COMPUTE,
+    DirectoryScopes.ADMIN_DIRECTORY_GROUP,
+    GenomicsScopes.GENOMICS,
+    "https://www.googleapis.com/auth/cloud-billing",
+    PlusScopes.USERINFO_EMAIL,
+    PlusScopes.USERINFO_PROFILE)
+
+  val credential: Credential = {
+    new GoogleCredential.Builder()
+      .setTransport(httpTransport)
+      .setJsonFactory(jsonFactory)
+      .setServiceAccountId(serviceAccountClientId)
+      .setServiceAccountScopes(saScopes.asJava) // grant bucket-creation powers
+      .setServiceAccountPrivateKeyFromPemFile(new java.io.File(serviceAccountPemFile))
+      .build()
+  }
+
+  private def billing: Cloudbilling = {
+    new Cloudbilling.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build()
+  }
+
+  def cloudResources: CloudResourceManager = {
+    new CloudResourceManager.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build()
+  }
+
+  def servicesManager: ServiceManagement = {
+    new ServiceManagement.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build()
+  }
+
+  def computeManager: Compute = {
+    new Compute.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build()
+  }
+
+  def storage: Storage = {
+    new Storage.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build()
+  }
+
+  def iam: Iam = {
+    new Iam.Builder(httpTransport, jsonFactory, credential).setApplicationName(appName).build()
+  }
+
+  def transferProjectOwnership(project: String, owner: String): Future[String] = {
+    /* NOTE: There is no work to be done here. It is up to the caller, inside their own FC stack to:
+     * - set up IAM policies in a manner similar to Rawls' addPolicyBindings
+     * - set up permissions on the Cromwell auth bucket
```

Comment:
the bullets in this comment are a lie! we do both of those things here. it is up to the caller to add the project to the rawls db and create a sam resource for it.